### PR TITLE
Added db.rel.query(type, func, options) and db.rel.find(type, options)

### PR DESCRIPTION
**Pull request by jkleinsc**
Right now there is no way to pass along options to PouchDB.allDocs from db.rel.find. This PR adds the ability to pass along options to allDocs.  I need this for paging, but I guess there may be other uses as well.

### CODE REVIEW

**Comment by jkleinsc** [comment]
@nolanlawson this isn't the PR for query that we talked about, but I think it adds value to be able to pass along options to allDocs from a find.

File: test/test.js [hunk 1]
```diff
@@ -1888,5 +1888,122 @@ function tests(dbName, dbType) {
...
+      }).then(function () {
+        return db.rel.save('post', {
+          title: 'Rails is Unagi',
+          text: 'Declicious unagi',
+          id: 2
+        });
+      }).then(function () {
+        return db.rel.find('post', null, {
+          endkey: db.rel.makeDocID({ type: 'post', id: 1 }),
+        });
```
Comment (nolanlawson) [inline test/test.js]: I was really hoping that people wouldn't have to get so deep into the implementation details like this...

**Comment by nolanlawson** [comment]
Have to admit I am kind of on the fence about this PR. I'm still trying to keep relational-pouch pretty tightly coupled to ember-pouch, and the Ember-y solution to this problem is to add a `query()` method. (`find()` with CouchDB-esque options would be definitely non-standard). That said, it's definitely a big limitation of the API not to be able to slice and dice data within types.

I think I would be more on board with this PR if it specified a subset of `allDocs()` options that could be passed in - e.g. just `limit`, `startkey`, and `endkey`. Also it should really do the ID manipulation for you; you should be able to do something like:

```
db.find('post', {
  startkey: 1,
  endkey: 3
});
```

That `null` is also unnecessary because you can do argument inference. If we made those changes, then I'd be +1 on the PR.

**Comment by jkleinsc** [comment]
@nolanlawson I made the changes we talked about yesterday.

**Comment by nolanlawson** [comment]
@jkleinsc OK, I will have a look this weekend. Sorry, just been very busy these past couple of days!

**Comment by jkleinsc** [comment]
@nolanlawson I decided to take a shot at adding db.rel.query too.  I didn't end up messing with emit and instead just skipped sorting the results from pouchdb.query.  I'm happy to adjust the code if it isn't satisfactory, just let me know.  Thanks!

**Comment by nolanlawson** [comment]
So this PR looks amazing; the tests are great and coverage is passing, so everything looks really good.

I love the new `find()` option, but I definitely think that `query()` needs some work. Ideally it should handle serialization of types (not filter in-memory), and I also think that `relational-pouch` users should be shielded from having to implement map functions (so basically, it should use `pouchdb-find` instead of map/reduce.

However since the other stuff is great, I'm happy to split this PR into two and open up the `query()` implementation for separate discussion. I'll go ahead and do that, and then let's move discussion to that issue.

**Comment by nolanlawson** [comment]
See #45, also you are now a maintainer, so cheers. :smiley: The rules are simple: don't break the build, wait for a +1 from another maintainer before merging, unless it's something minor like a README fix or publishing a new npm version. (Rules apply to me as well.)

**Pull request closed without merging**